api/addons/__init__.py

```diff
@@ -125,9 +121,16 @@ async def list_addons(
             }
             if details:
                 vinf["client_pyproject"] = await addon.get_client_pyproject()
-                vinf["client_source_info"] = await addon.get_client_source_info(
-                    base_url=base_url
-                )
+
+                source_info = await addon.get_client_source_info(base_url=base_url)
+                if source_info is None:
+                    pass
+
+                elif not all([isinstance(x, SourceInfo) for x in source_info]):
+                    logging.error(f"Invalid source info for {addon.name} {version}")
+                else:
+                    vinf["client_source_info"] = source_info
```

Comment:
There would be missing "client_source_info" key.
```suggestion
                    logging.error(f"Invalid source info for {addon.name} {version}")
                    source_info = []

                vinf["client_source_info"] = source_info
```